src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -767,24 +823,42 @@ void BLEManagerImpl::HandleTXCompleted(struct bt_conn * conId, void * /* param *
 
 void BLEManagerImpl::HandleConnect(struct bt_conn * conId, uint8_t err)
 {
-    ChipDeviceEvent event;
+    if (PlatformMgr().TryLockChipStack())
+    {
+        ChipDeviceEvent event;
 
-    event.Type                            = DeviceEventType::kPlatformZephyrBleConnected;
-    event.Platform.BleConnEvent.BtConn    = bt_conn_ref(conId);
-    event.Platform.BleConnEvent.HciResult = err;
+        // Don't handle BLE connecting events when it is not related to CHIPoBLE
+        VerifyOrExit(ConnectivityMgr().IsBLEAdvertisingEnabled(), );
 
-    PlatformMgr().PostEvent(&event);
+        event.Type                            = DeviceEventType::kPlatformZephyrBleConnected;
+        event.Platform.BleConnEvent.BtConn    = bt_conn_ref(conId);
+        event.Platform.BleConnEvent.HciResult = err;
+
+        PlatformMgr().PostEvent(&event);
+
+    exit:
+        chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+    }
 }
 
 void BLEManagerImpl::HandleDisconnect(struct bt_conn * conId, uint8_t reason)
 {
-    ChipDeviceEvent event;
+    if (PlatformMgr().TryLockChipStack())
```

Comment:
I'm a bit afraid of this use of `TryLockChipStack` as it may fail. I `LockChipStack` is not an option for some reason, then we should probably leave it as it was and add a task to our backlog to provide atomic `ConnectivityMgr().IsBLEAdvertisingEnabled()`.